### PR TITLE
i#1960 drltrace tests: added initial exit-code-only test for drltrace

### LINKED ISSUE #1960: tests for drltrace
We need a test for drltrace (probably similar to the test that we have for drstrace).

### PR DESCRIPTION
**Pull request by mxmssh**
Added initial exit-code-only test for drltrace.

### CODE REVIEW

File: drltrace/CMakeLists.txt [hunk 1]
```diff
@@ -109,3 +109,12 @@ endif ()
...
   PERMISSIONS ${owner_access} OWNER_EXECUTE GROUP_READ GROUP_EXECUTE
   WORLD_READ WORLD_EXECUTE)
+
+##################################################
+# drltrace test
+
+if (BUILD_TOOL_TESTS)
+  get_target_property(app_path drsyscall_app LOCATION${location_suffix})
+  get_target_property(drltrace_path drltrace LOCATION${location_suffix})
+  add_test(drltrace ${drltrace_path} -- ${app_path})
```
Comment (derekbruening) [inline drltrace/CMakeLists.txt]: All this will test is that the app didn't crash.  As a quick, initial test, sure, but we should have tests that ensure the output includes the library routines we expect drltrace to print.  Could we have comments here about adding more tests.  Also, the commit message should not say that it fixes #1960 as it is just one small exit-code test -- and thus it should also switch to the multi-commit title style (e.g., "i#1960 drltrace tests: add initial exit-code-only test").  (Remember to ensure the final squash-and-merge commit message is what we want -- editing the pull request title doesn't affect that.)

Comment (mxmssh) [inline drltrace/CMakeLists.txt]: done

Comment (mxmssh) [inline drltrace/CMakeLists.txt]: Required manual squash & merge

Comment (mxmssh) [inline drltrace/CMakeLists.txt]: There is a strange error at AppVeyor
Build started
git clone -q --depth=10 https://github.com/DynamoRIO/drmemory.git C:\projects\drmemory
fatal: early EOF
fatal: The remote end hung up unexpectedly
fatal: index-pack failed
error: RPC failed; curl 56 SSL read: error:00000000:lib(0):func(0):reason(0), errno 10054
Command exited with code 128

Comment (mxmssh) [inline drltrace/CMakeLists.txt]: Is that an internal error at AppVeyor or not?

Build started
git clone -q --depth=10 https://github.com/DynamoRIO/drmemory.git C:\projects\drmemory
git fetch -q origin +refs/pull/1964/merge:
fatal: unable to access 'https://github.com/DynamoRIO/drmemory.git/': Unknown SSL protocol error in connection to github.com:443 
Command exited with code 128

Comment (mxmssh) [inline drltrace/CMakeLists.txt]: Probably need to prepare empty commit to check that...

Comment (derekbruening) [inline drltrace/CMakeLists.txt]: Looks to me like some network issue between appveyor and github.

Comment (mxmssh) [inline drltrace/CMakeLists.txt]: So, submitting empty commit?

Comment (derekbruening) [inline drltrace/CMakeLists.txt]: There is at the appveyor page, I just clicked "rerun".  Might require appveyor admin privileges for this repo?

Comment (derekbruening) [inline drltrace/CMakeLists.txt]: But I don't see any changes to the code, only a merge from master, so you could have waited for when you commit your code changes and it will rerun then, right?

Comment (mxmssh) [inline drltrace/CMakeLists.txt]: right

**Comment by mxmssh** [comment]
It seems that we can merge it (I don't have permission).

**Comment by derekbruening** [comment]
I think you forgot to push your changes?  I only see merges from master.

File: drltrace/CMakeLists.txt (hunk 1 above)
Comment (mxmssh) [inline drltrace/CMakeLists.txt]: There is a change here.

Comment (derekbruening) [inline drltrace/CMakeLists.txt]: I see 3 commits: the original, and then 2 merges.  I thought the review requested changes and thus there would be another commit?

Comment (mxmssh) [inline drltrace/CMakeLists.txt]: You asked only to change commit message, there is no change for code. I have just added a single exit-only test for drltrace executable.

Comment (derekbruening) [inline drltrace/CMakeLists.txt]: Looking.  OK there is only one review comment (thought there were more, prob confusing this w/ the other PR's), but it does request a comment in the cmake for adding more tests (i.e., "XXX i#1960: ....").

nit: also note there is only one colon: https://github.com/DynamoRIO/dynamorio/wiki/Code-Reviews#commit-messages

File: drltrace/CMakeLists.txt [hunk 2]
```diff
@@ -109,8 +109,11 @@ install(TARGETS drltrace DESTINATION "${INSTALL_BIN}"
   WORLD_READ WORLD_EXECUTE)
 
 ##################################################
-# drltrace test
+# drltrace tests
 
+# XXX i#1960: add more tests for drltrace that checks the output contains
```
Comment (derekbruening) [inline drltrace/CMakeLists.txt]: nit: grammar: check

**Comment by derekbruening** [comment]
I have a plan for how to fix #1810 which will eliminate some of these appveyor failures

File: drltrace/CMakeLists.txt (hunk 2 above)
Comment (mxmssh) [inline drltrace/CMakeLists.txt]: I will fix it in some next commit.

**Comment by mxmssh** [comment]
hello.shadow test failed, so it requires manual squash & merge.
